### PR TITLE
[CN-343]-Refactoring-e2e-tests

### PR DESCRIPTION
**Pull request by cheels**

### CODE REVIEW

File: test/e2e/helpers_test.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+			seq = match[i]
+		}
+	}
+	if seq == "" {
+		Fail("Backup sequence not found")
+	}
+	return seq
+}
+
+func ReadLogs(scanner *bufio.Scanner, matcher types2.GomegaMatcher) {
```
Comment (SeriyBg) [inline test/e2e/helpers_test.go:65]: Do we really need this `func`? For me reading `EventyuallyInLogs().Should` is clearer than `ReadLogs()`

File: test/e2e/helpers_test.go [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+
+	hazelcastcomv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	n "github.com/hazelcast/hazelcast-platform-operator/controllers/naming"
+	"github.com/hazelcast/hazelcast-platform-operator/controllers/platform"
+	"github.com/hazelcast/hazelcast-platform-operator/test"
+)
+
+func GetBackupSequence(t time.Time) string {
+	By("Finding Backup sequence")
+	logs := test.GetPodLogs(context.Background(), types.NamespacedName{
```
Comment (SeriyBg) [inline test/e2e/helpers_test.go:38]: We can reuse `InitLogs` here, can't we?

**Review by SeriyBg — COMMENTED** [review]
Left a few comments that are rather minor

File: test/e2e/helpers_test.go (hunk 2 above)
Comment (cheels) [reply test/e2e/helpers_test.go:38]: Yes, good catch! 👍

File: test/e2e/helpers_test.go (hunk 1 above)
Comment (cheels) [reply test/e2e/helpers_test.go:65]: I don't know, for me, ReadLogs looks more compact without digging into details about timeout that we nowhere configuring in the tests. But with upcoming changes with Eventually timeout EventyuallyInLogs() might be better and more flexible.
Revert this to initial state

File: test/e2e/helpers_test.go [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+		log.Fatal(err)
+	}
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+	return clientSet
+}
+
+func ChangeHzClusterState(clusterState string, interval, timeout time.Duration) string {
```
[the remark this answers is not in the document]
Comment (cheels) [reply test/e2e/helpers_test.go]: Agree! I added ClusterState type

File: test/e2e/helpers_test.go [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+
+	hzClient "github.com/hazelcast/hazelcast-go-client"
+
+	hazelcastcomv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
+	n "github.com/hazelcast/hazelcast-platform-operator/controllers/naming"
+	"github.com/hazelcast/hazelcast-platform-operator/controllers/platform"
+	"github.com/hazelcast/hazelcast-platform-operator/test"
+)
+
+type ClusterState string
```
Comment (SeriyBg) [inline test/e2e/helpers_test.go:35]: We have the `ClusterState` type under `hazelcast`. Could we reuse it instead of redefining it?

**Pull request closed without merging**